script.js

```diff
@@ -0,0 +1,365 @@
+const API_URL = 'https://raw.githubusercontent.com/pahanan/pahan_an.github.io/refs/heads/main/api/data.json';
+const INGREDIENTS_API_URL = 'https://raw.githubusercontent.com/pahanan/pahan_an.github.io/refs/heads/main/api/ingredient-prices.json';
+let recipes = [];
```

Comment:
avoid using global variables for data this way. It can be confusing, especially if other variables have the same name. see https://www.geeksforgeeks.org/why-to-avoid-global-variables-in-javascript/